Funcky.Analyzer/Funcky.Analyzer.CodeFixes/EnumerableRepeatNeverCodeFix.cs

```diff
@@ -0,0 +1,71 @@
+            => ImmutableArray.Create(EnumerableRepeatNeverAnalyzer.DiagnosticId);
+
+        public override FixAllProvider GetFixAllProvider()
+            => WellKnownFixAllProviders.BatchFixer;
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            var diagnosticSpan = context.Diagnostics.First().Location.SourceSpan;
+            var declaration = root.FindToken(diagnosticSpan.Start).Parent.AncestorsAndSelf().OfType<InvocationExpressionSyntax>().First();
+
+            context.RegisterCodeFix(CreateFix(context, declaration), GetDiagnostic(context));
+        }
+
+        private static Diagnostic GetDiagnostic(CodeFixContext context)
+            => context.Diagnostics.First();
+
+        private CodeAction CreateFix(CodeFixContext context, InvocationExpressionSyntax declaration)
+            => CodeAction.Create(
+                EnumerableRepeatNeverCodeFixTitle,
+                CreateSequenceReturnAsync(context.Document, declaration),
+                nameof(EnumerableRepeatNeverCodeFixTitle));
+
+        private Func<CancellationToken, Task<Document>> CreateSequenceReturnAsync(Document document, InvocationExpressionSyntax declaration)
+            => async cancellationToken
+                => document.WithSyntaxRoot(await ReplaceWithSequenceReturn(document, declaration, cancellationToken).ConfigureAwait(false));
+
+        private async Task<SyntaxNode> ReplaceWithSequenceReturn(Document document, InvocationExpressionSyntax declaration, CancellationToken cancellationToken)
+        {
+            var oldRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+            var semanticModel = await document.GetSemanticModelAsync();
+
+            return oldRoot.ReplaceNode(declaration, CreateEnumerableReturnRoot(ExtractFirstArgument(declaration), semanticModel));
+        }
+
+        private static ArgumentSyntax ExtractFirstArgument(InvocationExpressionSyntax invocationExpr)
+            => invocationExpr.ArgumentList.Arguments[Argument.First];
+
+        private SyntaxNode CreateEnumerableReturnRoot(ArgumentSyntax firstArgument, SemanticModel model)
+            => InvocationExpression(
+                    MemberAccessExpression(
+                        SyntaxKind.SimpleMemberAccessExpression,
+                        IdentifierName(nameof(Enumerable)),
+                        GenericName(Identifier(nameof(Enumerable.Empty)))
+                        .WithTypeArgumentList(TypeArgumentList(SingletonSeparatedList(CreateTypeFromArgumentType(firstArgument, model))))));
+
+        private static TypeSyntax CreateTypeFromArgumentType(ArgumentSyntax firstArgument, SemanticModel model)
+            => ParseTypeName(model.GetTypeInfo(firstArgument.Expression).Type.ToDisplayString());
```

Comment:
`ToMinimalDisplayString` is the recommended solution for this since it is (if I understood correctly) context aware.
See: https://stackoverflow.com/questions/8231211/get-typesyntax-from-itypesymbol

```suggestion
            => ParseTypeName(model.GetTypeInfo(firstArgument.Expression).Type.ToMinimalDisplayString(model, firstArgument.SpanStart))
```